src/services/welcomeSocketService.ts

```
@@ -0,0 +1,102 @@
+
+    this.socket.on('game-status-changed', (data) => {
+      this.emit('game-status-changed', data);
+    });
+
+    this.socket.on('error', (data) => {
+      this.emit('error', data);
+    });
+  }
+
+  // Event system for React components
+  on(event: string, callback: Function) {
```

Biome (1.9.4)

[error] Don't use 'Function' as a type.

Prefer explicitly define the function shape. This type accepts any function-like value, which can be a common source of bugs.

(lint/complexity/noBannedTypes)